### PR TITLE
[qt6] a couple of QString updates

### PR DESCRIPTION
**Pull request by m-kuhn**
## Description

The QString api changed a bit for Qt6, prepare, prepare

### CODE REVIEW

File: src/app/main.cpp [hunk 1]
```diff
@@ -1434,7 +1434,7 @@ int main( int argc, char *argv[] )
         break;
       }
 
-      coords[i] = myInitialExtent.midRef( posOld, pos - posOld ).toDouble( &ok );
+      coords[i] = QStringView{myInitialExtent}.mid( posOld, pos - posOld ).toDouble( &ok );
```
Comment (nyalldawson) [inline src/app/main.cpp:1437]: do we need to explicitly use string view? I thought it was transparent...

Comment (m-kuhn) [reply src/app/main.cpp:1437]: Copied from here: https://doc-snapshots.qt.io/qt6-dev/qtcore-changes-qt6.html#the-qstringref-class

Comment (nyalldawson) [reply src/app/main.cpp:1437]: Ok, I have some learning to do :D

Comment (m-kuhn) [reply src/app/main.cpp:1437]:
```suggestion
      coords[i] = QStringView{myInitialExtent}.mid( posOld, pos - posOld ).toString().toDouble( &ok );
```

File: src/server/services/wms/qgswmsparameters.cpp [hunk 2]
```diff
@@ -570,7 +570,7 @@ namespace QgsWms
     const QRegExp composerParamRegExp( QStringLiteral( "^MAP\\d+:" ), Qt::CaseInsensitive );
     if ( key.contains( composerParamRegExp ) )
     {
-      const int mapId = key.midRef( 3, key.indexOf( ':' ) - 3 ).toInt();
+      const int mapId = QStringView{key}.mid( 3, key.indexOf( ':' ) - 3 ).toInt();
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsparameters.cpp:573]:
```suggestion
      const int mapId = QStringView{key}.mid( 3, key.indexOf( ':' ) - 3 ).toString().toInt();
```

File: src/app/main.cpp [hunk 3]
```diff
@@ -1443,7 +1443,7 @@ int main( int argc, char *argv[] )
 
     // parse last coordinate
     if ( ok )
-      coords[3] = myInitialExtent.midRef( posOld ).toDouble( &ok );
+      coords[3] = QStringView{myInitialExtent}.mid( posOld ).toDouble( &ok );
```
Comment (m-kuhn) [inline src/app/main.cpp:1446]:
```suggestion
      coords[3] = QStringView{myInitialExtent}.mid( posOld ).toString().toDouble( &ok );
```

File: tests/bench/main.cpp [hunk 4]
```diff
@@ -563,7 +563,7 @@ int main( int argc, char *argv[] )
         break;
       }
 
-      coords[i] = myInitialExtent.midRef( posOld, pos - posOld ).toDouble( &ok );
+      coords[i] = QStringView{myInitialExtent}.mid( posOld, pos - posOld ).toDouble( &ok );
```
Comment (m-kuhn) [inline tests/bench/main.cpp:566]:
```suggestion
      coords[i] = QStringView{myInitialExtent}.mid( posOld, pos - posOld ).toString().toDouble( &ok );
```

File: tests/bench/main.cpp [hunk 5]
```diff
@@ -572,7 +572,7 @@ int main( int argc, char *argv[] )
 
     // parse last coordinate
     if ( ok )
-      coords[3] = myInitialExtent.midRef( posOld ).toDouble( &ok );
+      coords[3] = QStringView{myInitialExtent}.mid( posOld ).toDouble( &ok );
```
Comment (m-kuhn) [inline tests/bench/main.cpp:575]:
```suggestion
      coords[3] = QStringView{myInitialExtent}.mid( posOld ).toString().toDouble( &ok );
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**